config/main.py

```diff
@@ -131,6 +131,21 @@ def reload(filename):
     command = "{} -j {} --write-to-db".format(SONIC_CFGGEN_PATH, filename)
     run_command(command, display_cmd=True)
     client.set(config_db.INIT_INDICATOR, True)
+    command = "{} -j {} -v \"DEVICE_METADATA['localhost']['hostname']\"".format(SONIC_CFGGEN_PATH, filename)
+    p = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE)
+    p.wait()
+    hostname = p.communicate()[0].strip()
+    _change_hostname(hostname)
+    run_command("service interfaces-config restart", display_cmd=True)
```

Comment:
Lines 139-148 duplicates lines 195-204. Please create a separate function (e.g., `restart_services()`) to contain these lines so they only need to be modified in one place.